### PR TITLE
Fix: Fix modified settings sync active file

### PR DESCRIPTION
**Pull request by MaShizhen**
- [LU-2122](https://snapmaker2.atlassian.net/browse/LU-2122)
- [LU-2150](https://snapmaker2.atlassian.net/browse/LU-2150)
- [LU-2147](https://snapmaker2.atlassian.net/browse/LU-2147)

### CODE REVIEW

File: src/app/flux/printing/index.js [hunk 1]
```diff
@@ -1855,17 +1864,18 @@ export const actions = {
 
     recordModelBeforeTransform: (modelGroup) => (dispatch) => {
         dispatch(operationHistoryActions.clearTargetTmpState(INITIAL_STATE.name));
+        const { recovery } = modelGroup.unselectAllModels();
         for (const model of modelGroup.selectedModelArray) {
-            const { recovery } = modelGroup.unselectAllModels();
+            modelGroup.unselectAllModels();
```
Comment (xiekun1992) [inline src/app/flux/printing/index.js:1869]: why you need to call `modelGroup.unselectAllModels();` in every loop

File: src/app/flux/printing/index.js [hunk 2]
```diff
@@ -1880,8 +1890,9 @@ export const actions = {
             }
         }
 
+        const { recovery } = modelGroup.unselectAllModels();
         for (const model of modelGroup.selectedModelArray) {
-            const { recovery } = modelGroup.unselectAllModels();
+            modelGroup.unselectAllModels();
```
Comment (xiekun1992) [inline src/app/flux/printing/index.js:1895]: why you need to call modelGroup.unselectAllModels(); in every loop

Comment (MaShizhen) [reply src/app/flux/printing/index.js:1895]: The `modelGroup.getSelectedModelTransformationForPrinting` is invoked in the loop, and the location information is obtained through the selected model.

File: src/server/services/api/api-profile-definitions.js [hunk 3]
```diff
@@ -177,7 +180,7 @@ export const updateDefinition = async (req, res) => {
 
     let filePath = '';
     let activeRecoverPath = '';
-    if (definitionId === 'snapmaker_extruder_0' || definitionId === 'snapmaker_extruder_1') {
+    if (definitionId === 'snapmaker_extruder_0' || definitionId === 'snapmaker_extruder_1' || definitionId === 'active' || definitionId === 'active_final') {
```
Comment (jane-rose) [inline src/server/services/api/api-profile-definitions.js:183]: use constant

**Review by jane-rose — APPROVED** [review]

**Review by wen-qing-255 — APPROVED** [review]

**Review by xiekun1992 — APPROVED** [review]

**Comment by jane-rose** [comment]
Leave LU-2150

**Pull request merged**